### ISSUE TITLE
Unable to select minDate & maxDate via keyboard navigation

### ISSUE THREAD

**Issue by WillSnowdon** [Hacker0x01/react-datepicker#2405]
**Describe the bug**

When providing minDate or maxDate properties I am unable to navigate to the actual minimum and maximum date in the date picker popup though I am able to click the dates.  The issue only occurs when only either a min or max is provided but not when both are.

**To Reproduce**

1. Go to https://reactdatepicker.com/
2. Click on maxDate example.
3. Attempt to navigate to the end date with keyboard
4. Unable to select last available date.

**Expected behavior**
You should be able to select the maximum available date with keyboard as is possible on click.

**Screenshots**
<img width="1275" alt="Screenshot 2020-09-24 at 10 18 27" src="https://user-images.githubusercontent.com/6881775/94126277-5333da00-fe4f-11ea-8e0d-e31bf37f002b.png">

**Desktop (please complete the following information):**

- OS: macOS
- Browser Chrome
- Version 85

**Smartphone (please complete the following information):**

**Additional context**

I've seen the issue occur with minDate but for some reason it behaves as expected on reactdatepicker.com min date example.  I'm unsure why, I've observed the issue occur with very similar settings locally. 

I've managed to work around the issue for now by always setting a very distant minDate if there is only a maxDate and vice versa.
